fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -278,6 +278,11 @@ class FullyShardedDataParallel(nn.Module):
             The `OffloadConfig` object is used to specify the type of offload (i.e SSD, CPU) and
             other required knobs when offloading parameters from GPU. Currently the OffloadConfig
             only supports specifying SSD offload as an option. Note: This is an experimental feature.
+        state_dict_on_rank_0_only (bool):
+            When set to ``True``, ``model.state_dict()`` will only returns full state dict on
```

Comment:
nit: "return a full state dict on rank 0 and an empty state dict on all other ranks." Another thing to mention here as well as in the comments is that this is to prevent OOMs which was the original reason for this change if I remember correctly.

Reply:
Added at the end :)